resources/splunk_app.rb

```diff
@@ -136,3 +148,41 @@ def config_scope
     apply_config
   end
 end
+
+class PackagedApp < SplunkApp
+  property :source_url, String, required: true
+
+  resource_name :splunk_app_package
+
+  def package_path
+    return @package_path if @package_path
+    file_cache = Pathname.new(Chef::Config['file_cache_path'])
+    @package_path = file_cache.join(CernerSplunk::PathHelpers.filename_from_url(source_url).gsub(/.spl$/, '.tgz'))
+  end
+
+  action :install do
+    return unless !version || changed?(:version)
+
+    remote_file package_path.to_s do
```

Comment:
I guess what I was asking is, why does it matter that we fail for local config in an app when the consumer is specifically trying to use that app.

Properly handling that with the future upgrade strategy would complicate things though. It would be better to fail on that case.